sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -1122,7 +1119,7 @@ case class StringSpace(child: Expression)
   """)
 // scalastyle:on line.size.limit
 case class Substring(str: Expression, pos: Expression, len: Expression)
-  extends TernaryExpression with ImplicitCastInputTypes {
+  extends TernaryExpression with ImplicitCastInputTypes with NullIntolerant {
```

Comment:
@nsyca If you have a bandwidth, could you please review all the expressions and see whether they can be marked as `NullIntolerant`? 

You can check the impl of these expressions and compare them with the corresponding ones in the other RDBMS. Thanks!